tasks/update-infra-deployments.yaml

```diff
@@ -9,19 +9,28 @@ spec:
       description: Script for changing the infra-deployments
     - name: ORIGIN_REPO
     - name: REVISION
+    - name: shared-secret
+      default: infra-deployments-pr-creator
   workspaces:
     - name: deploy-key
       optional: true
       description: |
         Must contain:
           'id_rsa' - deploy key for redhat-appstudio-mr-creation/infra-deployments/
           'appstudio-staging-ci' - private key for appstudio-staging-ci Github app
+  volumes:
+    - name: infra-deployments-pr-creator
+      csi:
+        readOnly: true
+        driver: csi.sharedresource.openshift.io
+        volumeAttributes:
+          sharedSecret: $(params.shared-secret)
   steps:
     - name: update-infra-deployments
       image: registry.redhat.io/openshift-pipelines/pipelines-git-init-rhel8@sha256:af7dd5b3b1598a980f17d5f5d3d8a4b11ab4f5184677f7f17ad302baa36bd3c1
-      env:
-        - name: WORKSPACE_SSH_DIRECTORY_PATH
-          value: $(workspaces.deploy-key.path)
+      volumeMounts:
+        - name: infra-deployments-pr-creator
+          mountPath: /tmp/infra-deployments-pr-creator
       script: |
         cp -R "${WORKSPACE_SSH_DIRECTORY_PATH}" ~/.ssh
```

Comment:
`${WORKSPACE_SSH_DIRECTORY_PATH}` has been removed, so it should be replaced by `/secrets/deploy-key`